### ISSUE TITLE
Readiness of application reported too early

### ISSUE THREAD

**Issue by rieckpil** [MicroShed/microshed-testing#110]
**Describe the bug**

Hey @aguibert, I wanted to create a blog post about this nice technology and started with a sample application (Jakarta EE + MP + PostgreSQL). The container setup is the following:

```
public class SampleApplicationConfig implements SharedContainerConfiguration {

    @Container
    public static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>()
            .withNetworkAliases("mypostgres")
            .withUsername("duke")
            .withPassword("duke42")
            .withDatabaseName("users");

    @Container
    public static MicroProfileApplication app = new MicroProfileApplication()
            .withEnv("POSTGRES_HOSTNAME", "mypostgres")
            .withEnv("POSTGRES_PORT", "5432")
            .withEnv("POSTGRES_USERNAME", "duke")
            .withEnv("POSTGRES_PASSWORD", "duke42")
            .withEnv("message", "Hello World from MicroShed Testing")
            .withAppContextRoot("/")
            .withReadinessPath("/resources/sample")
            //.withReadinessPath("/health/ready") reports readiness too early
            .dependsOn(postgres);

    @Override
    public void startContainers() {
        postgres.start();
        app.start();
    }

}
```

With the setup above it waits until the application is "ready" and my tests are green. But once I want to use the MicroProfile Health API to check readiness with `.withReadinessPath("/health/ready")` it reports readiness way too early right after:

```
15692 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKI0001I: The CORBA name server is now available at corbaloc:iiop:localhost:2809/NameService.
16716 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/metrics/
16716 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/openapi/ui/
16716 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/ibm/api/
16718 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/jwt/
16718 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/health/
16870 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/openapi/
20940 INFO org.microshed.testing.testcontainers.MicroProfileApplication  - STDOUT: [AUDIT   ] CWWKT0016I: Web application available (default_host): http://29f2c873a699:9080/
```

I'm not sure whether the setup is wrong or this is something with OpenLiberty and not your library. It seems like once the application starts the sequence of request "/health/ready" is the following: CONNECTION REFUSED -> 404 Liberty Page -> 200 OK (this is too early as the application is not ready) -> 200 OK with custom Health API check

**To Reproduce**

Find the project [here](https://github.com/rieckpil/blog-tutorials/tree/master/review-microshed-testing) and execute the `SampleApplicationIntegrationTest`

**Expected behavior**

It should wait until the application is started.

**System information:**
 - OS: Linux
 - Java Version: 11
 - MicroShed Testing Version: 0.5

**Additional context**

**Comment by aguibert** [comment]
HI @rieckpil, thanks for giving it a try! Yes this is a known bug OpenLiberty (see here: https://github.com/OpenLiberty/open-liberty/issues/9609) and should be fixed in a couple of weeks when OpenLiberty 19.0.0.12 is released.

In the meantime, the workaround is to not use the MP Health readiness check and instead use some application endpoint for readiness (e.g. create a dummy readiness JAX-RS endpoint)

I'm happy that you are thinking of writing on blog on MicroShed Testing too, that would be great! I see that your current sample has Liberty, and would like to draw your attention to the Liberty module (which is described in more details in [this guide](https://openliberty.io/guides/microshed-testing.html)). In particular, it uses a special Liberty extension module:
```xml
        <dependency>
            <groupId>org.microshed</groupId>
            <artifactId>microshed-testing-liberty</artifactId>
            <version>0.5</version>
            <scope>test</scope>
        </dependency>
```
which will allow the Liberty server to run in "dev mode" continually and just have MicroShed Testing reuse the already running instance (which is much faster for local development), but can still be run in "container mode" by running a different target (without making any code changes).

**Comment by rieckpil** [comment]
thanks for the quick reply @aguibert and hinting me to the Open Liberty bug. I also did the trick with using a JAX-RS dummy endpoint as an alternative and this works :+1: 

I'll update the example to use the MicroShed Liberty Testing dependency, thanks for the hint. 

PS: I closed this issue as this has nothing to do with MicroShed Testing
